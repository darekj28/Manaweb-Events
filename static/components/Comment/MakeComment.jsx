var React = require('react');
export default class MakeComment extends React.Component {
	constructor(props) {
		super(props);
		this.state = { canPost : true };
	}
	componentDidMount() {
		var messageVisible = true;
		var that = this;
		$('.post-button').click(function() {
			$(this).blur();
		});
		$('.post-button').focus(function() {
			$(this).blur();
		});
		$('#CommentPost').hide();
		$('#ToggleComment').click(function() {
			$('#CommentPost').slideToggle(function() {
				$('#CommentInput').focus();
			});
		})
	}
	handleCommentSubmit() {
		if (!this.state.canPost) swal("Yo!", "Please wait 10 seconds between comments.", "warning");
		else {
			if (this.commentText.value.trim().length > 0) {
				this.setState({ canPost : false, timeout : setTimeout(this.allowPost.bind(this), 10000) });
				this.props.onCommentSubmit(this.commentText.value);
			}
			else 
				swal("Oops...", "You can't post an empty message!", "error");
		}
	}
	allowPost() {
		clearTimeout(this.state.timeout);
		this.setState({ canPost : true });
	}
	handleCommentChange() {
		this.props.onCommentChange(this.commentText.value);
	}
	componentWillUnmount() {
		clearTimeout(this.state.timeout);
	}
	render() {
		return(
			<div id="MakeComment"> 
				<div id="ToggleComment" className="important-text">Reply to {this.props.op}...</div>
				<div id="CommentPost">
					<textarea id="CommentInput" className="form-control important-text" 
							value={this.props.commentText} 
							placeholder={this.props.placeholder} rows="4" ref={(input) => this.commentText = input} 
							onChange={this.handleCommentChange.bind(this)}></textarea>
					<div className="input-group-addon">
						<button className="btn post-button SubmitButton" onClick={this.handleCommentSubmit.bind(this)}><b>Comment</b></button>
					</div>
				</div>
			</div>);
	}
}