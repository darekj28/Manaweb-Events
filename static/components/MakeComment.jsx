var React = require('react');
// var $ = require('jquery');
export default class MakeComment extends React.Component {
	constructor(props) {
		super(props);
		this.handleCommentChange = this.handleCommentChange.bind(this);
		this.handleCommentSubmit = this.handleCommentSubmit.bind(this);
		this.handleEnterPress = this.handleEnterPress.bind(this);
	}
	componentDidMount() {
		var messageVisible = true;
		var that = this;
		$('#SubmitButtonComment').click(function() {
			$(this).blur();
		});
		$('#StartButtonComment').click(function(){
			$(this).blur();
	        if(!messageVisible){
	            messageVisible = !messageVisible;
	            $(this).siblings('#CommentPost').fadeIn(function(){
	                $(this).children('#CommentInput').focus();
	            });
	        } else {
	            messageVisible = !messageVisible;
	            $(this).siblings('#CommentPost').fadeOut(function(){
	                $(this).children('#CommentInput').blur();
	            });
	        } 
	    });  
	}
	handleCommentSubmit() {
		if (this.commentText.value.length > 0)
			this.props.onCommentSubmit(this.commentText.value);
	}
	handleCommentChange() {
		this.props.onCommentChange(this.commentText.value);
	}
	handleEnterPress(target) {
		if (this.commentText.value.length > 0)
		    if (target.charCode==13)
		        this.props.onCommentSubmit(this.commentText.value); 
	}
	render() {
		return(
			<div> 
				<a id="StartButtonComment" className="StartButton">
					<span className="glyphicon glyphicon-pencil AppGlyphicon"></span>
				</a>
				<div id="CommentPost" className="Message pull-right input-group input-group-unstyled">
					<input id="CommentInput" type="text" className="CommentText form-control" 
							onKeyPress={this.handleEnterPress} value={this.props.commentText} 
							placeholder={this.props.placeholder} ref={(input) => this.commentText = input} 
							onSubmit={this.handleCommentSubmit} onChange={this.handleCommentChange}></input>
				
					<a className="SubmitButton input-group-addon" id="SubmitButtonComment"
							onClick={this.handleCommentSubmit}>
						<span className="glyphicon glyphicon-send AppGlyphicon"></span>
					</a>
				
				</div>
			</div>);
	}
}