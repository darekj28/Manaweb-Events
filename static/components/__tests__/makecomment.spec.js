import MakeComment from "../MakeComment.jsx";

describe("MakeComment", ()=>{
	it("renders correctly", () => {
		const actions = ["Trade", "Play", "Chill"];
		const mc = renderer.create(
			<MakeComment commentText="a" onClick={jest.fn()} onCommentChange={jest.fn()}
				onCommentSubmit={jest.fn()} actions={actions}/>
			);
		expect(mc).toMatchSnapshot();
	});
	it("should call handleCommentChange with text in textbox when user types", () => {
		expect(true).toBe(true);
	});
	it("should call handleCommentSubmit with text in textbox when user presses submit or enter", () => {
		expect(true).toBe(true);
	});
});