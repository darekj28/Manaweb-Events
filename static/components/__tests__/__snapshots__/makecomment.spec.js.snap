exports[`MakeComment renders correctly 1`] = `
<div>
  <a
    className="StartButton"
    id="StartButtonComment">
    <span
      className="glyphicon glyphicon-pencil AppGlyphicon" />
  </a>
  <div
    className="Message pull-right input-group input-group-unstyled"
    id="CommentComment">
    <input
      className="CommentText form-control"
      id="CommentTextComment"
      onChange={[Function]}
      onKeyPress={[Function]}
      onSubmit={[Function]}
      placeholder={undefined}
      type="text"
      value="a" />
    <a
      className="SubmitButton input-group-addon"
      id="SubmitButtonComment"
      onClick={[Function]}>
      <span
        className="glyphicon glyphicon-send AppGlyphicon" />
    </a>
  </div>
</div>
`;
